### ISSUE TITLE
False negative: 'aria-required-children' rule does not ensure all the child elements have an appropriate role

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#2576]
```aria-required-children``` checks only if at least one child has the appropriate role rather than making sure all the children of a particular role have the appropriate roles

For example if an element with role ```list``` has one child with role ```listitem``` and other children with inappropriate roles, ```aria-required-children``` does not fail on these children

Expectation: Fail all child elements with inappropriate roles. Case1 and Case3 in the example should fail

Actual: The rule checks only if at least on child has the appropriate role

Demo link:  https://codepen.io/iamrafan/pen/qBNawdg

<pre><code>axe-core version: 4.0.2
axe version: 4.6.1</code></pre>

**Comment by WilcoFiers** [comment]
@iamrafan Hey Guru. ARIA doesn't require this. The description of required owned elements is somewhat problematic. I have been trying to get the ARIA working group to update their spec to clarify on this issue https://github.com/w3c/aria/issues/1033. As of today, The requirement is that at least one descendant (jup, doesn't even have to be a child) has the required role. It's pretty busted, and I hope they can get around to fixing this, but I don't want to change how axe-core works without seeing how the ARIA working group thinks it should be.
